### PR TITLE
[21.01] Submit input_format to tools API in payload, not as a parameter

### PR DESCRIPTION
**Pull request by simonbray**
A small fix to https://github.com/galaxyproject/galaxy/pull/10720.

### CODE REVIEW

File: lib/galaxy/webapps/galaxy/api/tools.py [hunk 1]
```diff
@@ -557,7 +554,7 @@ def _create(self, trans: GalaxyWebTransaction, payload, **kwd):
         # as a regular tool parameter we accept both.
         use_cached_job = payload.get('use_cached_job', False) or util.string_as_bool(inputs.get('use_cached_job', 'false'))
 
-        input_format = kwd.get('input_format', 'legacy')
+        input_format = payload.get('input_format', 'legacy')
```
Comment (simonbray) [inline lib/galaxy/webapps/galaxy/api/tools.py]:
```suggestion
        input_format = payload.get('input_format', '21.01')
```
A bit unrelated - this wasn't discussed in the original PR for some reason, but maybe it makes sense that '21.01' is actually the default value in Galaxy version 21.01?

Comment (nsoranzo) [reply lib/galaxy/webapps/galaxy/api/tools.py]: I guess the reason to keep `legacy` is default is that Galaxy API is generally very conservative in trying not to break backward-compatibility.

**Comment by nsoranzo** [comment]
I've rebased to target `release_21.01`, which I think was the intention. Also I've restored the `input_format` documentation, I think it's fine even if it's passed through the payload.

**Review by nsoranzo — APPROVED** [review]

File: lib/galaxy/webapps/galaxy/api/tools.py (hunk 1 above)
Comment (mvdbeek) [reply lib/galaxy/webapps/galaxy/api/tools.py]: That and the frontend doesn't submit in this format.

File: lib/galaxy/webapps/galaxy/api/tools.py [hunk 2]
```diff
@@ -559,7 +563,7 @@ def _create(self, trans, payload, **kwd):
         # as a regular tool parameter we accept both.
         use_cached_job = payload.get('use_cached_job', False) or util.string_as_bool(inputs.get('use_cached_job', 'false'))
 
-        input_format = kwd.get('input_format', 'legacy')
+        input_format = str(payload.get('input_format', 'legacy'))
```
Comment (mvdbeek) [inline lib/galaxy/webapps/galaxy/api/tools.py:566]:
```suggestion
        input_format = payload.get('input_format', 'legacy')
```

Comment (mvdbeek) [reply lib/galaxy/webapps/galaxy/api/tools.py:566]: Ah, saw the commit message. We should fix that at the root.

**Pull request merged**